nion/swift/Inspector.py

```diff
@@ -588,109 +588,97 @@ def __property_changed_from_graphic(self, name: str) -> None:
+        self._caption_current_index = Model.PropertyModel[int](0)
+        self._session_id_model = Model.PropertyChangedPropertyModel[str](display_item, "session_id")
+        self._created_local_as_string_model = Model.PropertyChangedPropertyModel[str](display_item, "created_local_as_string")
+        self.info_title_label: typing.Optional[UserInterface.Widget] = None
+
+        u = Declarative.DeclarativeUI()
+
+        TOOL_TIP_STR = _("Use empty field for automatic title.")
+
+        self.ui_view = u.create_column(
+            u.create_row(
+                u.create_label(text=_("Title"), width=60, tooltip=TOOL_TIP_STR),
+                u.create_line_edit(name="info_title_label", text="@binding(_title_model.value)", placeholder_text="@binding(_placeholder_title_model.value)", tooltip=TOOL_TIP_STR),
+                u.create_spacing(8)
+            ),
+            u.create_row(
+                u.create_column(
+                    u.create_label(text=_("Caption"), width=60),
+                    u.create_stretch()
+                ),
+                u.create_stack(
+                    u.create_column(
+                        u.create_text_edit(height=60, editable=False, text="@binding(_caption_model.value)"),
+                        u.create_row(
+                            u.create_push_button(text=_("Edit"), on_clicked="_begin_caption_edit"),
+                            u.create_stretch()
+                        ),
+                    ),
+                    u.create_column(
+                        u.create_text_edit(height=60, text="@binding(_editable_caption_model.value)"),
+                        u.create_row(
+                            u.create_push_button(text=_("Save"), on_clicked="_save_caption_edit"),
+                            u.create_push_button(text=_("Cancel"), on_clicked="_end_caption_edit"),
+                            u.create_stretch()
+                        )
+                    ),
+                    current_index="@binding(_caption_current_index.value)"
+                ),
+                u.create_spacing(8)
+            ),
+            u.create_row(
+                u.create_label(text=_("Session"), width=60),
+                u.create_label(text="@binding(_session_id_model.value)", width=240),
+                u.create_stretch()
+            ),
+            u.create_row(
+                u.create_label(text=_("Date"), width=60),
+                u.create_label(text="@binding(_created_local_as_string_model.value)"),
+            ),
+            spacing=4
+        )
+
+    def _begin_caption_edit(self, widget: UserInterface.Widget) -> None:
+        self._editable_caption_model.value = self._display_item.caption
+        self._caption_current_index.value = 1
+
+    def _save_caption_edit(self, widget: UserInterface.Widget) -> None:
+        new_caption = self._editable_caption_model.value
+        self._display_item.caption = new_caption if new_caption is not None else str()
```

Comment:
`self._display_item.caption = ...`: should go through `DisplayItemPropertyCommandModel` to make it undoable.

Additional comment: with these changes, I'm ok with this code; but there might be a more streamlined way to do the editing and updating rather than using two text area widgets. After this gets merged, we might want to revisit it.

Reply:
Changing the caption directly was an oversight on my part and is not fixed.

I had a similar thought about streamlining it while I was working on it, but decided to maintain the current process for now.